### PR TITLE
Addition of Desec provider

### PR DESCRIPTION
**Pull request by ainsey11**
As discussed in #12 

This implements deSEC as a provider, using the new style provider support framework

sadly the deSEC go library isn't the nicest to work with, and I don't have any spare domains I can register over there to test with and have been struggling for time with work commitments,  but I've tried to be as thorough as I can with the tests,

I won't merge yet as I'd like to add the documentation in, but would like a review to make sure we're happy with the code first

### CODE REVIEW

**Comment by pier-oliviert** [comment]
I had a quick look at it and it's looking really good. I'm trying to build a custom github action to take care of all the unique need we have for this project. One of which is the ability to create multiple packages for a given release. Hopefully, I'll have it ready by Friday so we can get some cool releases with providers.

All of this to say, good work!

Now that you have used this new approach, I am curious how it felt to use it the first time?

**Comment by ainsey11** [comment]
Thanks!

It was quite nice once I got to grips with it and used it in a hands on nature, 

The type checks when we add the server code to ensure the provider has the create/delete functions works really nicely and saved me from a few silly mistakes 

I think I can see a few inefficiencies in the docker builds but it's really nothing major and I don't want to change too much until we've got the next release out, but I can see we're using the same builder steps repeated, docker catching should handle that but it's just a lot of repetitive code 

On the whole though it's a really nice solution, and I was able to get the desec provider knocked out in a couple of hours, it took me longer to fight the tests than it did to implement!

**Comment by pier-oliviert** [comment]
I've finally managed to change github action to do the sensible thing. I'm also working on a new github action that will help with the Helm chart, but I don't think that will impact you at all with the work you're doing. Now, if you want, you can add this provider to the github action and a new package will be generated alongside the others: https://github.com/pier-oliviert/phonebook/blob/main/.github/workflows/controller.yaml#L67-L77

There's probably a bigger conversation we'll need to have on how to manage all provider and how versioning should work with each providers. Right now, I anchored all versions with Phonebook's as it's much easier to reason about...

I hope I didn't derail your work too much here :)

**Comment by ainsey11** [comment]
Thanks @pier-oliviert - looks good to me :) 

Simple pull from master and merge sorted everything out, I've updated the PR with the new builds and added some documentation in, it's limited but out of all the providers we have, it's one of the simpler ones so hopefully fairly user-safe

Would be great if I could get a final review before I merge
Thanks!

**Review by pier-oliviert — COMMENTED** [review]
I think this is almost good to go. The few `PB-DESEC-#00xx` errors you have don't have a matching entry in errors.md, so adding those lines, then I'll merge it!

Great work.

File: Dockerfile.controller
Comment (ainsey11) [reply Dockerfile.controller:113]: good catch! I'd indeed messed up my copy and paste

**Comment by ainsey11** [comment]
Thanks for the review! Minor fixes added and docs now all updated :)

**Review by pier-oliviert — APPROVED** [review]

**Pull request merged**
